packages/core/src/generateProductionServer.ts

```diff
@@ -138,20 +156,40 @@ function generateHelperFucntions(): ts.Node {
       })
     }
 
-    function writeResponse(response, res) {
-      res.writeHead(response.status, {
-        "Content-Type": "application/json",
-        "Access-Control-Allow-Origin": "*",
-        "Access-Control-Allow-Methods": "POST",
-        "Access-Control-Allow-Headers": "content-type",
-      })
+    async function writeResponse(originWhitelist, responseOrLazyResponse, res, req) {
+      res.setHeader("Content-Type", "application/json")
+
+      if (originWhitelist && originWhitelist.length > 0 && req.headers.origin !== undefined) {
+        const isValidOrigin = originWhitelist.includes(req.headers.origin)
+        
+        if (!isValidOrigin) {
+          res.statusCode = 401;
+          res.end(JSON.stringify({ error: "Invalid origin" }));
```

Comment:
Can we add something about "cors" or something to this error?

Reply:
Ah good one!